### PR TITLE
Update custom YAML loader tests to properly test unicode literals

### PR DESCRIPTION
**Pull request by terminalmage**
This PR adds tests for #42435. It also modifies the existing tests for readability using `textwrap.dedent()`.

### CODE REVIEW

File: tests/unit/utils/test_yamlloader.py [hunk 1]
```diff
@@ -89,15 +88,27 @@ def test_yaml_duplicates(self):
         Test that duplicates still throw an error
         '''
         with self.assertRaises(ConstructorError):
-            self._render_yaml(b'''
-p1: alpha
-p1: beta''')
+            self._render_yaml('''\
```
Comment (rallytime) [inline tests/unit/utils/test_yamlloader.py]: Does this section need `textwrap.dedent` as well?

File: tests/unit/utils/test_yamlloader.py [hunk 2]
```diff
@@ -89,15 +88,27 @@ def test_yaml_duplicates(self):
...
 
         with self.assertRaises(ConstructorError):
-            self._render_yaml(b'''
-p1: &p1
-  v1: alpha
-p2:
-  <<: *p1
-  v2: beta
-  v2: betabeta''')
+            self._render_yaml('''\
```
Comment (rallytime) [inline tests/unit/utils/test_yamlloader.py]: Same question here as above.

**Review by rallytime — COMMENTED** [review]
@terminalmage I had a question about two of the tests you've modified here. I am not sure why some tests needed the `textwrap.dedent` change, but not others.

**Review by rallytime — APPROVED** [review]

File: tests/unit/utils/test_yamlloader.py (hunk 2 above)
Comment (terminalmage) [inline tests/unit/utils/test_yamlloader.py]: Good call on both counts. Fixed.

**Comment by rallytime** [comment]
Test failures unrelated.

**Pull request merged**